File: Compilers编译原理实验/语言认知实验/Java/src/matrix.java
public class matrix{
    public static void main(String[] args) {
    // 初始化两个矩阵
    int size = 100;
    int [][] matrix_A = new int[size][size];
    int [][] matrix_B = new int[size][size];
    int [][] matrix_C = new int[size][size];
    
    //两个矩阵赋值
    for(int i = 0; i < size ; i++)
    {
        for(int j = 0;j < size; j++)
        {
            matrix_A[i][j] = (int)(Math.random()*(10));
            matrix_B[i][j] = (int)(Math.random()*(10));
            // System.out.println(matrix_A[i][j]);
        }
        System.out.println("\n");
    }
    long start_time = System.currentTimeMillis();    
    //两个矩阵相乘
    for(int i = 0; i < size ; i++)
    {
        for(int j = 0; j < size ; j++)
        {
            for(int k = 0; k < size ; k++)
            {
                matrix_C[i][j] += matrix_A[i][k] * matrix_B[k][j];
            }
        }
    }
    // for(int i = 0; i < size ; i++)
    // {
    //     for(int j = 0; j < size ; j++)
    //     {
    //         System.out.println(matrix_C[i][j]);
    //     }
    // }
    long end_time = System.currentTimeMillis();
    long time = end_time - start_time ;
    System.out.println("Java cost time: " + time + "ms");
    }
}

File: Compilers编译原理实验/语言认知实验/Haskell/src/matrix.hs
import Data.List
import System.CPUTime
import Text.Printf

multiplyMatrix :: Num x => [[x]] -> [[x]] -> [[x]]
multiplyMatrix matrix_A matrix_B = [ [ sum $ zipWith (*) matrix_A_row matrix_B_col | matrix_B_col <- (transpose matrix_B) ] | matrix_A_row <- matrix_A ]


getMatrix n =
  let rowIndex n k = [x | x <- [1,2..n]]
  in [rowIndex n k | k <- [1,2..n]]

main = do
  let size = 100::Int
  --初始化1000*1000的矩阵
  let matrix_A = getMatrix size 
  let matrix_B = getMatrix size

  start_time <- getCPUTime
  let matrix_C = multiplyMatrix matrix_A matrix_B
  print matrix_C
  end_time <- getCPUTime
  let totalTime = (fromIntegral(end_time - start_time)) / (10 ^ 9)
  printf "Haskell cost time: %f ms\n" (totalTime :: Double)
